packages/amplify-graphql-conversation-transformer/src/resolvers/invoke-lambda-resolver.ts

```diff
@@ -14,27 +16,42 @@ export const invokeLambdaMappingTemplate = (
   config: ConversationDirectiveConfiguration,
   ctx: TransformerContextProvider,
 ): MappingTemplateProvider => {
-  const req = createInvokeLambdaRequestFunction(config, ctx);
-  const res = createInvokeLambdaResponseFunction(config);
-  return MappingTemplate.inlineTemplateFromString(dedent(req + '\n' + res));
+  const { TOOL_DEFINITIONS_LINE, TOOLS_CONFIGURATION_LINE } = generateToolLines(config);
+  const SELECTION_SET = selectionSet;
+  const GRAPHQL_API_ENDPOINT = ctx.api.graphqlUrl;
+  const MODEL_CONFIGURATION_LINE = generateModelConfigurationLine(config);
+  const RESPONSE_MUTATION_NAME = config.responseMutationName;
+  const RESPONSE_MUTATION_INPUT_TYPE_NAME = config.responseMutationInputTypeName;
+  const MESSAGE_MODEL_NAME = config.messageModel.messageModel.name.value;
+
+  const substitutions = {
+    TOOL_DEFINITIONS_LINE,
+    TOOLS_CONFIGURATION_LINE,
+    SELECTION_SET,
+    GRAPHQL_API_ENDPOINT,
+    MODEL_CONFIGURATION_LINE,
+    RESPONSE_MUTATION_NAME,
+    RESPONSE_MUTATION_INPUT_TYPE_NAME,
+    MESSAGE_MODEL_NAME,
+  };
+
+  let resolver = fs.readFileSync(path.join(__dirname, 'invoke-lambda-resolver-fn.js'), 'utf8');
+  Object.entries(substitutions).forEach(([key, value]) => {
+    const replaced = resolver.replace(new RegExp(`\\[\\[${key}\\]\\]`, 'g'), value);
+    resolver = replaced;
+  });
+
+  // This unfortunately needs to be an inline template because an s3 mapping template doesn't allow the CDK
+  // to substitute token values, which is necessary for this resolver function due to its reference of
+  // `ctx.api.graphqlUrl`.
+  return MappingTemplate.inlineTemplateFromString(resolver);
```

Comment:
In a follow up, I'm planning on moving the `ctx.api.graphqlUrl` reference to the "Resolver code" portion of the pipeline along with the existing `ctx.stash.` `typeName` / `fieldName` / etc values.

This will allow us to make this an S3 asset and support hotswapping for schema defined values like system prompt, tools, model (LLM), and model inference configuration.